js-intro.js

```diff
@@ -56,7 +92,28 @@ var testString = "thisisateststring"
 var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
 
 // 6a. Write a function that takes in the toonimals array as an argument and returns an array with only the toon objects that are cats. Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
-
+var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+cats =(...toonimals) => {
+  const {animal} = toonimals
+  for (let i = 0; i < toonimals.length; i++)
+var array = {}
+  if(animal === "cat"){
+    return array
+  }else{
+    return "hi"
+  }
+}
+console.log(cats(toonimals));
 //6b. Write a function that returns only the names of all the non-cats. Expected output: "Itchy" "Daffy" "Ren"
-
+var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+cats =(...toonimals) => {
+  const {animal} = toonimals
+  for (let i = 0; i < toonimals.length; i++)
+  if(animal!=== "cat"){
+    return array
+  }else{
+    return "hi"
+  }
+}
+console.log(cats(toonimals));
```

Comment:
Hey Josh, look above and see if you can solve this problem. Make sure to look up any methods or high-order function you need clarification on